### DISCUSSION TITLE
Port Hopping using XMUX

### DISCUSSION THREAD

**Discussion by fodhelper** [XTLS/Xray-core#4120] [Ideas]
Hello
XMUX has a great feature, it's 'source port hopping' and can bypass some ISP restrictions
But i am wondering why Xray Outbound does not accept multiple Ports and Port Ranges to also do a remote port hopping?

**Comment by tomxiang1** [comment]
The port range jumping feature is very obvious. XRAY mainly simulates normal browsers Internet behavior to avoid blocking.

**Comment by RPRX** [comment]
> multiple Ports and Port Ranges to also do a remote port hopping

~~我也觉得特征太明显了~~

**Reply by fodhelper** [reply]
It depend on how the GFW of different countries work, we don't know the exact Ability and Rules of each one
Using random port for each connection may even bypass some of them

But for GFW of China, i believe multiple IP:Port for outbound would work well, now that XHTTP supports separated downstream and upstream, if each stream have multiple different IP:Port it will even work better, it's good for both CDN and Direct connection to a server with multiple IPs
